app/services/container_dashboard_service.rb

```diff
@@ -149,22 +157,75 @@ def heatmaps
     }
   end
 
-  def ems_utilization
+  def heatmaps
+    hourly_heatmaps || realtime_heatmaps
+  end
+
+  def fill_ems_utilization(m, time, used_cpu, used_mem, total_cpu, total_mem)
+    used_cpu[time] += m.v_derived_cpu_total_cores_used if m.v_derived_cpu_total_cores_used.present?
+    used_mem[time] += m.derived_memory_used if m.derived_memory_used.present?
+    total_cpu[time] += m.derived_vm_numvcpus if m.derived_vm_numvcpus.present?
+    total_mem[time] += m.derived_memory_available if m.derived_memory_available.present?
+  end
+
+  def realtime_ems_utilization
+    used_cpu = Hash.new(0)
+    used_mem = Hash.new(0)
+    total_cpu = Hash.new(0)
+    total_mem = Hash.new(0)
+
+    realtime_provider_metrics.each do |m|
+      minute = m.timestamp.beginning_of_minute.utc
+      fill_ems_utilization(m, minute, used_cpu, used_mem, total_cpu, total_mem)
+    end
+
+    {
+      :interval_name => "realtime",
+      :xy_data       => ems_utilization_data(used_cpu, total_cpu, used_mem, total_mem) || {:cpu => nil, :mem => nil}
+    }
+  end
+
+  def hourly_ems_utilization
```

Comment:
the `realtime_ems_utilization` and `hourly_ems_utilization` methods look suspiciously the same, can you do one function that take the `time_frame` (hour or minute) as a parameter ?